browser-test/src/admin_application_statuses.test.ts

```diff
@@ -114,16 +119,43 @@ describe('view program statuses', () => {
 
     describe('when a status is changed, a confirmation dialog is shown', () => {
       it('when rejecting, the selected status is not changed', async () => {
-        await adminPrograms.setStatusOptionAndDismissModal(statusName)
+        await adminPrograms.setStatusOptionAndAwaitModal(noEmailStatusName)
+        await dismissModal(adminPrograms.applicationFrame())
         expect(await adminPrograms.getStatusOption()).toBe('Choose an option:')
       })
 
       it('when confirmed, the page is redirected with a success toast', async () => {
-        await adminPrograms.setStatusOptionAndConfirmModal(statusName)
+        const modal = await adminPrograms.setStatusOptionAndAwaitModal(
+          noEmailStatusName,
+        )
+        await adminPrograms.confirmStatusUpdateModal(modal)
         expect(await adminPrograms.getStatusOption()).toBe('Choose an option:')
         await adminPrograms.expectUpdateStatusToast()
         // TODO(#3020): Assert that the selected status has been updated.
       })
+
+      it('when no email is configured for the status, a warning is shown', async () => {
+        const modal = await adminPrograms.setStatusOptionAndAwaitModal(
+          noEmailStatusName,
+        )
+        expect(await modal.innerText()).toContain(
```

Comment:
Do we have a test the the email checkbox is shown when there is an email to be sent?

Reply:
Nope, which is what the TODO tracks. We can't currently add a unit test since we can't test the positive logged-in case from the controller.

We also can't do a browser test since we don't support logging in as an applicant (and thus configuring an email).